lib/fetch/request.js

```diff
@@ -421,7 +421,7 @@ class Request {
       // list, append header’s name/header’s value to this’s headers.
       if (headers.constructor.name === 'Headers') {
         for (const [key, val] of headers[kHeadersList] || headers) {
-          this[kHeaders].append(key, val)
+          this[kState].headersList.append(key, val)
```

Comment:
Why this change?

Reply:
Without it, doing something along the lines of 
```js
import { fetch, Headers } from 'undici'

const headers = new Headers([['cookie', 'value']])

await fetch('http://example.com', { headers })
```

wouldn't work. My best guess is removing the filtered headers but I'm not entirely sure. I believe I also made a similar change for the same reason.

Reply:
I'm iffy about landing this before we understand what's going on here. Last time we were in This situation I found a rather serious bug.

Reply:
I think it's mostly confusing with having multiple Headers/HeadersList for each Request and confusion with distinguishing one from another.

https://github.com/nodejs/undici/blob/0e6427467e5c2d604dca32b7cb01c0b4979d25de/lib/fetch/request.js#L421
The spec says to use "this's headers" and undici uses `this[kHeaders]`.

https://github.com/nodejs/undici/blob/0e6427467e5c2d604dca32b7cb01c0b4979d25de/lib/fetch/request.js#L427
Then a few lines down the spec says to also fill "this's Headers" and `this[kState].headersList` is being used.

Without looking too far into it, removing `this[kHeaders]` would probably be beneficial, and use `this[kState].headersList` (maybe change it to a Headers instance?) instead. Or remove one of them, so there's less confusion on which should be used?

Reply:
With looking into it, we can set `this[kHeaders]` to a Headers class and set `request.headersList` to `this[kHeaders].headersList`?

Reply:
Using `.set` and `.append` on the `Headers` instance doesn't seem to update its HeadersList. Updating the list itself, however, does work.

Reply:
I'm open for changes that are consistent and follow the wording of the spec.

Reply:
Found a solution and was able to remove the weird hack that was added last time this happened.